### DISCUSSION TITLE
LocalStorage || Execute Script

### DISCUSSION THREAD

**Discussion by CaReS0107** [spatie/browsershot#788] [Bugs]
Is there a way to add local storage data? Or is there a way to execute a script with browsershot?

**Comment by cas-technology** [comment]
As you can render HTML as-is, you should be able to execute javascript. Try it out with an executing function, like:
`
<script>
function foo() {
  alert('bar');
}
foo();
</script>
`
As for local storage, I won't pretend to know.

**Reply by CaReS0107** [reply]
@cas-technology now sure what you mean, the main idea is to add token into local storage to set authentication for puppeteer
